Trappist/src/Promact.Trappist.Core/Controllers/ProfileController.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Promact.Trappist.DomainModel.Models;
+using Promact.Trappist.Repository.Profile;
+using Promact.Trappist.Web.Models;
+using System.Threading.Tasks;
+
+namespace Promact.Trappist.Core.Controllers
+{
+    [Authorize]
+    [Route("api/profile")]
+    public class ProfileController : Controller
+    {
+        private readonly IProfileRepository _profileRepository;
+        private readonly SignInManager<ApplicationUser> _signInManager;
+
+        public ProfileController(IProfileRepository profileRepository, SignInManager<ApplicationUser> signInManager)
+        {
+            _profileRepository = profileRepository;
+            _signInManager = signInManager;
+        }
+
+        /// <summary>
+        /// Gets current user profile details
+        /// </summary>
+        /// <returns>User profile details</returns>
+        [HttpGet]
+        public async Task<IActionResult> GetUserDetails()
+        {
+            var user = await _profileRepository.GetUserDetailsAsync(User.Identity.Name);
+            return Ok(user);
+        }
+
+        /// <summary>
+        /// Update current user profile details
+        /// </summary>
+        /// <param name="updateUserDetails">takes parameter of type ApplicationUser which comes from the client side(from body)</param>
+        /// <returns>Update user profile details in database</returns>
+        [HttpPut]
+        public IActionResult UpdateProfile([FromBody]ApplicationUser updateUserDetails)
+        {
+            if (ModelState.IsValid)
+            {
+                _profileRepository.UpdateUserProfile(updateUserDetails);
+                return Ok();
+            }
+            return BadRequest();
+        }
+
+        /// <summary>
+        /// Update user password 
+        /// </summary>
+        /// <param name="model">takes the parameter of type ChangePasswordModel</param>
+        /// <returns>save the new password in the database</returns>
+        [Route("password")]
+        [HttpPut]
+        public async Task<IActionResult> ChangePassword([FromBody]ChangePasswordModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                ModelState.AddModelError("error", "Enter details Correctly");
+                return BadRequest(ModelState);
+            }
+            var result=await _profileRepository.UpdateUserPasswordAsync(User.Identity.Name,model);
+            if (!result)
+            {
+                ModelState.AddModelError("error", "Your current password is wrong");
```

Comment:
result will only be false when the current password is wrong. In all other cases client side  validation will be executed. And if the format of the password is wrong then that will be checked by ModelState.isValid. If it is invalid it will return badrequest withe the message as writtenin the code.